### PR TITLE
[image_picker_for_web] Migrate image_picker to package:cross_file

### PR DESCRIPTION
**Pull request by BeMacized**
Changed all usages of PickedFile to XFile, to follow the changes made to the platform interface (#4072)

This PR will remain in draft state until the platform interface changes have been merged and published (#4072), so that the pubspec can be updated properly.

Relevant issue:
- flutter/flutter#70886

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Comment by ditman** [comment]
Let's go! The platform_interface has been published here:

https://pub.dev/packages/image_picker_platform_interface/changelog#220

**Comment by ditman** [comment]
@googlebot I consent.

**Comment by ditman** [comment]
@BeMacized, I've brought the branch up to date with the latest `master`, removed the "deprecated" methods (let's add those only in the core plugin package), and used this opportunity to migrate the tests to run with `flutter drive`, instead of `flutter test --platform=chrome`.

There's documentation linked in the new README files, but TL;DR, this is how tests can be run now:

1. Ensure `chromedriver --port=4444` is running
2. From the root of the repo, run: `dart ./script/tool/lib/src/main.dart drive-examples --packages=image_picker/image_picker_for_web --web`.

  ```
dit@dit:~/github/plugins$ dart ./script/tool/lib/src/main.dart drive-examples --packages=image_picker/image_picker_for_web --web
script/tool/lib/src/main.dart: Warning: Interpreting this as package URI, 'package:flutter_plugin_tools/src/main.dart'.

============================================================
|| Running for image_picker_for_web
============================================================

Running command: "flutter drive -d web-server --web-port=7357 --browser-name=chrome --driver test_driver/integration_test.dart --target integration_test/image_picker_for_web_test.dart" in /usr/local/google/home/dit/github/plugins/packages/image_picker/image_picker_for_web/example
Running "flutter pub get" in example...                            614ms
Launching integration_test/image_picker_for_web_test.dart on Web Server in debug mode...
Waiting for connection from debug service on Web Server...         16.6s
integration_test/image_picker_for_web_test.dart is being served at http://localhost:7357
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge devices for an improved development workflow.
Application finished.
[WARNING]: http://localhost:7357/dart_sdk.js 48732:16 "registerExtension() from dart:developer is only supported in build/run/test environments where the developer event method hooks have been set."
[WARNING]: http://localhost:7357/dart_sdk.js 48714:16 "postEvent() from dart:developer is only supported in build/run/test environments where the developer event method hooks have been set."
[INFO]: http://localhost:7357/dart_sdk.js 27220:14 "00:00 +0: Can select a file (Deprecated)"
[WARNING]: http://localhost:7357/packages/image_picker_for_web/image_picker_for_web.dart.js 204 File chooser dialog can only be shown with a user activation.
[INFO]: http://localhost:7357/dart_sdk.js 27220:14 "00:00 +1: Can select a file"
[WARNING]: http://localhost:7357/packages/image_picker_for_web/image_picker_for_web.dart.js 204 File chooser dialog can only be shown with a user activation.
[INFO]: http://localhost:7357/dart_sdk.js 27220:14 "00:00 +2: computeCaptureAttribute"
[INFO]: http://localhost:7357/dart_sdk.js 27220:14 "00:00 +3: createInputElement accept: any, capture: null"
[INFO]: http://localhost:7357/dart_sdk.js 27220:14 "00:00 +4: createInputElement accept: any, capture: something"
[INFO]: http://localhost:7357/dart_sdk.js 27220:14 "00:00 +5: (tearDownAll)"
[INFO]: http://localhost:7357/dart_sdk.js 27220:14 "00:00 +6: All tests passed!"
All tests passed.


------------------------------------------------------------
Run overview:
  image_picker_for_web - ran

Ran for 1 package(s)


No issues found!
  ```

**Comment by ditman** [comment]
_Just verified that tests are running in CI. See [this run](https://cirrus-ci.com/task/5019375276130304?logs=drive#L370), for example._

**Comment by BeMacized** [comment]
@ditman Alright! I can confirm the tests succeed locally as well, so I've marked this PR as ready for review.

File: packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart [hunk 1]
```diff
@@ -150,6 +212,26 @@ class ImagePickerPlugin extends ImagePickerPlatform {
...
   }
 
+  Future<XFile> _getSelectedXFile(html.FileUploadInputElement input) {
+    final Completer<XFile> _completer = Completer<XFile>();
+    // Observe the input until we can return something
+    input.onChange.first.then((event) {
+      final objectUrl = _handleOnChangeEvent(event);
+      if (!_completer.isCompleted && objectUrl != null) {
+        _completer.complete(XFile(objectUrl));
+      }
```
Comment (ditman) [inline packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart:222]: Let's extract the name on disk of the file from this `event` so we can initialize the XFile with it!

Comment (ditman) [reply packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart:222]: There's an example of it here: https://github.com/flutter/plugins/blob/master/packages/file_selector/file_selector_web/lib/src/dom_helper.dart#L59-L66

**Review by ditman — APPROVED** [review]
We'll leave the name/length extraction for v2.2.1, so we can finish this migration without too much feature creep.

**Comment by ditman** [comment]
Published https://pub.dev/packages/image_picker_for_web/changelog#210

**Pull request merged**